### PR TITLE
Code refactor

### PR DESCRIPTION
**Pull request by Debas-31**
- [x] Created a new class console and extracted function from main.rb 
- [x] Move method starts from App class to console.

### CODE REVIEW

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
### I will leave a few comments in this message because the lines which need to be adjusted haven't been changed in this pr.
- [ ] In the `create_student` and `create_teacher` method you are asking the user twice for their name and age this is not in line with [the DRY principle](https://www.itexico.com/blog/software-development-kiss-yagni-dry-3-principles-to-simplify-your-life) please try to avoid writing the same code twice.
- [ ] in the `create_student` and `create_teacher` methods you are also printing to the console, getting information from the user and saving the data received, this is not in line with the [single responsibility rule](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid), please refactor these methods so they comply with the rules.
https://github.com/Debas-31/school-library-OOP/blob/7d85937ab4a1fe63b94ce44260a7d49ea0dd9e75/app.rb#L41-L72
- [ ] in the `create_book` method you are also printing to the console, getting information from the user and saving the data received, this is not in line with the [single responsibility rule](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid), please refactor these methods so they comply with the rules.
https://github.com/Debas-31/school-library-OOP/blob/7d85937ab4a1fe63b94ce44260a7d49ea0dd9e75/app.rb#L74-L83
- [ ] in the `create_rental` method you are also printing to the console, getting information from the user and saving the data received, this is not in line with the [single responsibility rule](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid), please refactor these methods so they comply with the rules.
https://github.com/Debas-31/school-library-OOP/blob/7d85937ab4a1fe63b94ce44260a7d49ea0dd9e75/app.rb#L85-L105
- [ ] The link to the License is not working as expected, please have a look at [the template](https://github.com/microverseinc/readme-template) provided by Microverse for an example of how it is intended to work.
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb [hunk 1]
```diff
@@ -4,128 +4,40 @@
 require './book'
 require './classroom'
 require './rental'
+require './lib/create_books'
+require './lib/create_persons'
+require './lib/create_rental'
+require './lib/displayer'
 
 class App
```
Comment (FrederickMih) [inline app.rb:12]: - [ ] Hi @Debas-31, this class still has `multiple` responsibilities as pointed out by the previous reviewer. Remember the Single Responsibility Principle of **SOLID** states that a class should have `one and only one` reason to change that's a class should have `only` a single rule to play. In this class, in some of the methods, you are still carrying out multiple responsibilities like:
- printing data to the console( check lines 16, 20, 29, 34, 39, 48, 53, 54, 56, 58, 64, 68, 73, 74, 76, 78, 82, 86, 87, 89, 93, 97, 102, 109, 115, 119, 122).
- Getting information from the user (lines 18, 40, 55, 57, 75, 77, 88, 90, 100, 107, 110, 120) and
- Saving data.
This is not in line with the [SRP of SOLID](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid). If you are able to achieve the SRP, the possibility of reusing this class will increase :smiley: .

File: app.rb [hunk 2]
```diff
@@ -4,128 +4,40 @@
...
   end
 
   def list_all_persons
-    puts 'Database is empty! Add a person.' if @persons.empty?
-    @persons.each { |person| puts "[#{person.class.name}] Name: #{person.name} Age: #{person.age} Id: #{person.id}" }
+    @displayer.display_persons(@persons)
   end
 
   def create_person
-    print 'To create a student, press 1, to create a teacher, press 2 :'
```
Comment (FrederickMih) [inline app.rb:39]: - [ ] Asking the user for the initial choice of options( or to quit) should have been done in the main.rb file. Please let's fix this.

File: app.rb [hunk 3]
```diff
@@ -4,128 +4,40 @@
...
-  def create_student
-    puts 'Create a new student'
-    print 'Enter name: '
-    name = gets.chomp
-    print 'Enter student age: '
-    age = gets.chomp.to_i
-    print 'Has parent permission? [Y/N]: '
-    parent_permission = gets.chomp.downcase
-    case parent_permission
-    when 'n'
```
Comment (FrederickMih) [inline app.rb:61]: - [ ] Please let's make our codes simple (KISS). All requests for inputs and input validations should have been placed somewhere else or in main.rb file. Also, remember about the best practice and SOLID lessons.

File: app.rb [hunk 4]
```diff
@@ -4,128 +4,40 @@
...
-      student = Student.new(age, 'classroom', name, parent_permission: false)
```
Comment (FrederickMih) [inline app.rb:62]: - [Optional] This should have been done in the `create_person()` method

File: app.rb [hunk 5]
```diff
@@ -4,128 +4,40 @@
...
-      @persons << student
-      puts "Student doesn't have parent permission, can't rent a books "
-    when 'y'
-      student = Student.new(age, 'classroom', name, parent_permission: true)
```
Comment (FrederickMih) [inline app.rb:66]: - [Optional] This should have been done in the `create_person()` method

File: app.rb [hunk 6]
```diff
@@ -4,128 +4,40 @@
...
-  end
-
-  def create_teacher
-    puts 'Create a new teacher'
-    print 'Enter name: '
-    name = gets.chomp
-    print 'Enter teacher age'
-    age = gets.chomp.to_i
-    print 'Enter teacher specialization: '
-    specialization = gets.chomp
```
Comment (FrederickMih) [inline app.rb:79]: - [ ] All requests for inputs and input validations should have been placed somewhere else or in main.rb file.

File: app.rb [hunk 7]
```diff
@@ -4,128 +4,40 @@
...
-    teacher = Teacher.new(age, specialization, name)
```
Comment (FrederickMih) [inline app.rb:80]: [Optional] This should have been done in the `create_person()` method

File: app.rb [hunk 8]
```diff
@@ -4,128 +4,40 @@
...
 
-  def create_book
-    puts 'Create a new book'
-    print 'Enter title: '
-    title = gets.chomp
-    print 'Enter an author: '
-    author = gets.chomp
-    book = Book.new(title, author)
-    @books.push(book)
-    puts "Book #{title} created successfully."
```
Comment (FrederickMih) [inline app.rb:93]: - [ ] All requests for inputs and input should have been placed somewhere else or in main.rb file.

File: app.rb [hunk 9]
```diff
@@ -4,128 +4,40 @@
...
+  def create_book()
+    @books << CreateBook.new.create_book
   end
 
   def create_rental
-    puts 'Select which book you want to rent by entering its number'
```
Comment (FrederickMih) [inline app.rb:97]: - [ ] Asking the user for the initial choice of options( or to quit) should have been done in the main.rb file. Please let's fix this.

File: app.rb [hunk 10]
```diff
@@ -4,128 +4,40 @@
...
-    @books.each_with_index { |book, index| puts "[#{index}] Title: #{book.title} Author: #{book.author}" }
-
-    book_id = gets.chomp.to_i
-
-    puts 'Select a person from the list by its number'
```
Comment (FrederickMih) [inline app.rb:102]: - [ ] Asking the user for the initial choice of options( or to quit) should have been done in the main.rb file. Please let's fix this.

File: app.rb [hunk 11]
```diff
@@ -4,128 +4,40 @@
...
-
-    rental = Rental.new(date, @persons[person_id], @books[book_id])
-    @rentals << rental
-
-    puts 'Rental created successfully'
+    @rentals << CreateRental.new.create_rental(@books, @persons)
   end
 
   def list_all_rentals
-    print 'To see person rentals enter the person ID: '
```
Comment (FrederickMih) [inline app.rb:119]: - [ ] Asking the user for the initial choice of options( or to quit) should have been done in the main.rb file. Please let's fix this.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @Debas-31 , 

Great work on making the changes requested by a previous reviewer 👏 



You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 11 above)
Comment (Mithi-code) [reply app.rb:119]: These are the deleted code

File: app.rb (hunk 10 above)
Comment (Mithi-code) [reply app.rb:102]: These are the deleted code

File: app.rb (hunk 9 above)
Comment (Mithi-code) [reply app.rb:97]: These are the deleted code

File: app.rb (hunk 8 above)
Comment (Mithi-code) [reply app.rb:93]: These are the deleted code

File: app.rb (hunk 7 above)
Comment (Mithi-code) [reply app.rb:80]: These are the deleted code

File: app.rb (hunk 5 above)
Comment (Mithi-code) [reply app.rb:66]: These are the deleted code

File: app.rb (hunk 4 above)
Comment (Mithi-code) [reply app.rb:62]: These are the deleted code

File: app.rb (hunk 3 above)
Comment (Mithi-code) [reply app.rb:61]: These are the deleted code

File: app.rb (hunk 2 above)
Comment (Mithi-code) [reply app.rb:39]: These are the deleted code

File: app.rb (hunk 1 above)
Comment (Mithi-code) [reply app.rb:12]: We have made changes according to last reviewer and you checked our deleted code

File: app.rb (hunk 6 above)
Comment (Mithi-code) [reply app.rb:79]: These are the deleted code

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 @Debas-3, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**